src/watchmaker/status/__init__.py

```diff
@@ -37,11 +38,34 @@ def initialize(self, config=None):
             return
 
         status_provider_ids = []
-        if status_config.get_cloud_identifiers(config):
-            cloud_identifier = provider()
+        # get supported providers with prereqs
+        supported_providers = status_config.get_sup_cloud_ids_w_prereqs(config)
+
+        if supported_providers:
+            # Detect Provider
+            cloud_identifier = provider(supported_providers)
             if cloud_identifier != AbstractStatusProvider.identifier:
                 status_provider_ids.append(cloud_identifier)
 
+        if not status_provider_ids:
+            # Supported provider doesn't exist or not detected
+            # Check for supported providers missing prereqs
+            missing_prereq_cloud_ids = (
+                status_config.get_req_cloud_ids_wo_prereqs(config)
+            )
+
+            if missing_prereq_cloud_ids:
+                # Detect provider
+                cloud_identifier = provider(missing_prereq_cloud_ids)
+
+                if cloud_identifier != AbstractStatusProvider.identifier:
```

Comment:
I kinda hate the deeply nested `if` statements. Any easy ideas on something a little flatter or more declarative?

Reply:
Check the latest and see if you like better.

Reply:
🎆